Mathlib/Util/AssertExists.lean

```diff
@@ -49,4 +49,7 @@ You should *not* delete the `assert_not_exists` statement without careful discus
 -/
 elab "assert_not_exists " n:ident : command =>
   do let [] ← try resolveGlobalConstWithInfos n catch _ => return
-      | throw <| Exception.error n m!"Declaration {n} is not allowed to exist in this file"
+      | throw <| .error n <| m!"Declaration {n} is not allowed to be imported by this file.\n\n" ++
+          "These invariants are maintained by `assert_not_exists` statements,\n" ++
+          "and exist in order to ensure that \"complicated\" parts of the library are not" ++
+          "accidentally introduced as dependencies of \"simple\" parts of the library."
```

Comment:
```suggestion
      | throw <| .error n <| m!"Declaration {n} is not allowed to be imported by this file.\n\n\
          These invariants are maintained by `assert_not_exists` statements, \
          and exist in order to ensure that \"complicated\" parts of the library are not \
          accidentally introduced as dependencies of \"simple\" parts of the library."
```

Reply:
It looks like `\` at the end of a line doesn't work. I'm going to revert to `++`.

Reply:
@urkud That backslash at the end of the line wasn't a mistake. I forgot that this feature has not made it into the Lean release yet. It's just in the development version.

I'll push a working version.